plugins/colorschemes/kanagawa-paper.nix

```diff
@@ -0,0 +1,230 @@
+{
+  lib,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "kanagawa-paper";
+  isColorscheme = true;
+  packPathName = "kanagawa-paper.nvim";
+  package = "kanagawa-paper-nvim";
+
+  description = ''
+    You can select the theme in two ways:
+    - Set `colorschemes.kanagawa-paper.settings.theme` AND explicitly unset `vim.o.background` (i.e. `options.background = ""`).
+    - Set `colorschemes.kanagawa-paper.settings.background` (the active theme will depend on the value of `vim.o.background`).
+  '';
+
+  maintainers = [ lib.maintainers.FKouhai ];
+
+  # TODO introduced 2024-03-15: remove 2024-05-15
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "compile"
+    "undercurl"
+    "comments"
+    "functions"
+    "keyword"
+    "statement"
+    "type"
+    "transparent"
+    "dim_inactive"
+    "terminal_colors"
+    [
+      "colors"
+      "palette"
+    ]
+    [
+      "colors"
+      "theme"
+    ]
+    "theme"
+    [
+      "background"
+      "dark"
+    ]
+    [
+      "background"
+      "light"
+    ]
+
+  ];
+  imports = [
+    (lib.mkRemovedOptionModule
+      [
+        "colorschemes"
+        "kanagawa-paper"
+        "overrides"
+      ]
+      ''
+        Use `colorschemes.kanagawa-paper.settings.overrides` but you now have to add the full function definition:
+        ```
+          function(colors)
+            ...
+          end
+        ```
+      ''
+    )
+  ];
```

Comment:
If this is a new plugin, why do we have deprecation warnings? Unless I'm missing something, these can go.